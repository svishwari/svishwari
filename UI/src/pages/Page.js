import React from 'react'
import { useLocation } from 'react-router-dom'

const Page = () => {
    const location = useLocation();
    const currentPath = location.pathname.split("/").map(word => (word.charAt(0).toUpperCase() + word.slice(1))).join(" ").trim()
    return (
        <div>
            {currentPath}
        </div>
    )
}

export default Page
