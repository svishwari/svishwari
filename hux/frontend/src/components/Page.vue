<template>
  <div>
    <slot name="header" />
    <div class="container" :style="{ 'max-width': maxWidth, padding: padding }">
      <slot />
    </div>
    <slot name="footer" />
  </div>
</template>

<script>
export default {
  name: "Page",

  props: {
    maxWidth: {
      type: String,
      required: false,
    },
    padding: {
      type: String,
      required: false,
    },
  },
}
</script>

<style lang="scss" scoped>
.container {
  max-width: 1220px;
  padding: 30px;
  height: calc(100vh - 180px);
  overflow-y: auto !important;
  overflow-x: hidden !important;
}
</style>
