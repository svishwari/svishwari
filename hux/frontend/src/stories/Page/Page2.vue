<template>
  <div>
    <slot name="header" />
    <div
      class="container no-shadow"
      :style="{ 'max-width': maxWidth, padding: padding }"
    >
      <empty-page v-if="errorState" class="mt-n3" type="sad-face-2" :size="40">
        <template #title>
          <div class="text-h4 mt-4 error-text-color">Unavailable</div>
        </template>
        <template #subtitle>
          <div class="text-body-1 mt-4 error-text-color">
            Our team is working hard to fix it. <br />Please be patient and try
            again soon!
          </div>
        </template>
      </empty-page>
      <empty-page
        v-else-if="emptyState"
        type="trash_can"
        class="mt-n3"
        :size="40"
      >
        <template #title>
          <div class="text-h4 mt-4 empty-text-color">Welcome to the void</div>
        </template>
        <template #subtitle>
          <div class="text-body-1 mt-4 empty-text-color">
            You currently have no data.
          </div>
        </template>
      </empty-page>
      <slot />
    </div>
    <slot name="footer" />
  </div>
</template>

<script>
import EmptyPage from "@/components/common/EmptyPage.vue"

export default {
  name: "Page2",

  components: {
    EmptyPage,
  },

  props: {
    maxWidth: {
      type: String,
      required: false,
      default: "auto",
    },
    padding: {
      type: String,
      required: false,
      default: "auto",
    },
    emptyState: {
      type: Boolean,
      required: false,
      default: false,
    },
    errorState: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
}
</script>

<style lang="scss" scoped>
.container {
  max-width: 1220px;
  padding: 30px;
  height: calc(100vh - 180px);
  overflow-y: auto !important;
  overflow-x: hidden !important;
  box-shadow: -100px 0px 200px rgba(30, 30, 30, 0.03),
    -8px 0px 16px rgba(0, 85, 135, 0.15);
}
.empty-text-color {
  color: var(--v-black-lighten6) !important;
  font-weight: 400 !important;
}
.error-text-color {
  color: var(--v-black-base) !important;
  font-weight: 400 !important;
}
</style>
