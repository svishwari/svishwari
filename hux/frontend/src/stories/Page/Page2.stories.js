import Page2 from "./Page2.vue"

export default {
  component: Page2,
  title: "NewComponents/Page2",

  argTypes: {
    maxWidth: { control: "text" },
    padding: { control: "text" },
    errorState: { control: "boolean" },
    emptyState: { control: "boolean" },
  },

  args: {
    maxWidth: "auto",
    padding: "auto",
    errorState: true,
    emptyState: false,
  },

  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/eCWUinup52AoHQw1FBVJkd/HDS-(Hux-Design-System)?node-id=9409%3A68520",
    },
  },
}

const Template = (args, { argTypes }) => ({
  components: { Page2 },
  props: Object.keys(argTypes),
  template: `
  <page2 v-bind="$props" />`,
})

export const newPage = Template.bind({})
