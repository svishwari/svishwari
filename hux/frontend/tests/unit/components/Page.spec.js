import Page from "@/components/Page"
import { shallowMount } from "@vue/test-utils"
import Vuetify from "vuetify"

const vuetify = new Vuetify()

describe("Page", () => {
  test("Should have content in header slot", async () => {
    const headerSlotContent = "Header slot"
    const wrapper = shallowMount(Page, {
      slots: {
        header: `<div>${headerSlotContent}</div>`,
      },
    })
    expect(wrapper.text()).toContain(headerSlotContent)
    expect(wrapper).toMatchSnapshot()
  })
  test("Should have content in footer slot", async () => {
    const footerSlotContent = "Footer slot"
    const wrapper = shallowMount(Page, {
      slots: {
        footer: `<div>${footerSlotContent}</div>`,
      },
    })
    expect(wrapper.text()).toContain(footerSlotContent)
    expect(wrapper).toMatchSnapshot()
  })
})
