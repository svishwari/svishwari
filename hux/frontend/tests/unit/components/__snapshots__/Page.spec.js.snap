// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Should have content in footer slot 1`] = `
<div>
  <div class="container"></div>
  <div>Footer slot</div>
</div>
`;

exports[`Page Should have content in header slot 1`] = `
<div>
  <div>Header slot</div>
  <div class="container"></div>
</div>
`;
